test/test_image_warp.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+import time
+import unittest
+import numpy as np
+
+from tinygrad.tensor import Tensor
+from tinygrad.engine.jit import TinyJit
+from tinygrad.device import Device
+
+
+MODEL_WIDTH = 512
+MODEL_HEIGHT = 256
+W, H = 1928, 1208
+
+
+def warp_perspective_tinygrad(src, M_inv):
+  w_dst, h_dst = (MODEL_WIDTH, MODEL_HEIGHT)
+  w_src, h_src = (W,H)
+
+  x = Tensor.arange(w_dst).reshape(1, w_dst).expand(h_dst, w_dst)
+  y = Tensor.arange(h_dst).reshape(h_dst, 1).expand(h_dst, w_dst)
+  ones = Tensor.ones_like(x)
+  dst_coords = x.reshape(1, -1).cat(y.reshape(1, -1)).cat(ones.reshape(1, -1))
+
+  src_coords = M_inv @ dst_coords
+  src_coords = src_coords / src_coords[2:3, :]
+  x_src = src_coords[0].reshape(h_dst, w_dst)
+  y_src = src_coords[1].reshape(h_dst, w_dst)
+
+  x_nn_clipped = x_src.round().clip(0, w_src - 1).cast('int')
+  y_nn_clipped = y_src.round().clip(0, h_src - 1).cast('int')
+
+  idx = (y_nn_clipped * w_src + x_nn_clipped).reshape(-1)
+
+  src_flat = src.reshape(h_src * w_src)
+  sampled = src_flat[idx]
+  return sampled
+
+def warp_perspective_numpy(src, M_inv):
+  w_dst, h_dst = (MODEL_WIDTH, MODEL_HEIGHT)
+  h_src, w_src = src.shape[:2]
+  xs, ys = np.meshgrid(np.arange(w_dst), np.arange(h_dst))
+  dst_x = xs.reshape(-1)
+  dst_y = ys.reshape(-1)
+
+  ones = np.ones_like(xs)
+  dst_hom = np.stack([xs, ys, ones], axis=0).reshape(3, -1)
+
+  src_hom = M_inv @ dst_hom
+  src_hom /= src_hom[2:3, :]
+
+  src_x = np.clip(np.round(src_hom[0, :]).astype(int), 0, w_src - 1)
+  src_y = np.clip(np.round(src_hom[1, :]).astype(int), 0, h_src - 1)
+
+  dst = np.zeros((h_dst, w_dst), dtype=src.dtype)
+  dst[dst_y, dst_x] = src[src_y, src_x]
+  return dst.ravel()
+
+@unittest.skipIf(Device.DEFAULT == "WEBGPU", "Precision error")
+class TestImageWarpJit(unittest.TestCase):
+  update_img_jit = TinyJit(warp_perspective_tinygrad, prune=True)
```

Comment:
This shouldn't be in the class, it should be in a function in the class starting with `test_`